lib/src/value/color.ts

```diff
@@ -30,102 +37,77 @@ export class SassColor extends Value {
   private lightnessInternal?: number;
   private readonly alphaInternal: number;
 
-  private constructor(color: RgbColor | HslColor) {
+  constructor(color: RgbColor | HslColor | HwbColor) {
```

Comment:
This can be defined as constructor overloads to enforce the type of each param.

```suggestion
  constructor(color: RgbColor);
  constructor(color: HslColor);
  constructor(color: HwbColor);
  constructor(color: RgbColor | HslColor | HwbColor) {
```

(the same goes for the rest of these overloaded constructors and functions)

Reply:
Oh interesting, I didn't realize that was different from a type system perspective.

Reply:
using constructor overloads there is exactly the opposite of the typescript best practice.

And it does not provide any benefit. The first argument can indeed be either `RgbColor` or `HslColor` or `HwbColor` and that's exactly what the union type tells to the compiler.

Reply:
If this is just a union type, the compiler wouldn't complain about this:

```ts
new SassColor({
  red: 255,
  green: 255,
  blue: 255,
  hue: 0,
  saturation: 0,
  lightness: 100,
})
```

whereas overloads properly catch the error.

Reply:
Are you sure about that ? From the doc of typescript, the 2 cases should be equivalent

Reply:
I opened https://github.com/microsoft/TypeScript/issues/46873 to get the feedback from the TS team on that case.

Reply:
Note that this works only when using an object literal in the argument. As soon as the object comes from elsewhere (a variable being used), extra properties will be allowed (as that would be a subtype of the RgbColor type).
If you actually want to forbid passing `lightness` in a RgbColor (which might make sense in that context), you would have to define those HSL properties with type `never` in the RgbColor type.

Reply:
I think the pass-through case is compelling enough to just make this a union type. We can add `never` fields in a follow-up PR to make the types mutually exclusive.